llvm/lib/MC/GOFFObjectWriter.cpp

```diff
@@ -258,6 +699,268 @@ void GOFFObjectWriter::writeHeader() {
   OS.write_zeros(6);       // Reserved
 }
 
+void GOFFObjectWriter::writeSymbol(const GOFFSymbol &Symbol,
+                                   const MCAsmLayout &Layout) {
+  uint32_t Offset = 0;
+  uint32_t Length = 0;
+  GOFF::ESDNameSpaceId NameSpaceId = GOFF::ESD_NS_ProgramManagementBinder;
+  Flags SymbolFlags;
+  uint8_t FillByteValue = 0;
+
+  Flags BehavAttrs[10] = {};
+  auto setAmode = [&BehavAttrs](GOFF::ESDAmode Amode) {
```

Comment:
for I still prefer self document as I comment in https://reviews.llvm.org/D152945?id=531449#inline-1489119 ,

if you do define the BehavAttrs as I suggest.

you do not the lambda function to set the value , you assign the value directly.

Reply:
Doing it this way means the code matches the documentation. For example, BehavAttrs[3] would match the documented behavioral attributes at offset "3.x".